Add benchmark organization guide (#39)
aai-institute/nnbench: docs/guides/organization.md

# How to efficiently organize benchmark code

> To efficiently organize benchmarks and keeping your setup modular, you can follow a few guidelines.

Reviewer: ```suggestion
To efficiently organize benchmarks and keeping your setup modular, we propose the following guidelines.
```

## Tip 2: Group benchmarks by common attributes

To maintain good organization within your benchmark directory, you can group similar benchmarks into their own Python files.
As an example, if you have a set of benchmarks to establish data quality, and benchmarks for scoring trained models on curated data, you could structure them as follows:

```
benchmarks/
├── data_quality.py
├── model_perf.py
└── ...
```

This is helpful when running multiple benchmark workloads separately, as you can just point your benchmark runner to each of these separate files:

```python
from nnbench.runner import BenchmarkRunner

runner = BenchmarkRunner()
data_metrics = runner.run("benchmarks/data_quality.py", params=...)
# same for model metrics, where instead you pass benchmarks/model_perf.py.
model_metrics = runner.run("benchmarks/model_perf.py", params=...)
```

## Tip 3: Attach tags to benchmarks for selective filtering

For structuring benchmarks within files, you can also use **tags**, which are tuples of strings attached to a benchmark:

```python
# benchmarks/data_quality.py
import nnbench

> @nnbench.benchmark(tags=("foo",))

Reviewer: Should we add a small note like:

> Note the trailing comma. This is necessary for Python to correctly parse single entry tuples as such. Without it, the value will be interpreted as a string, causing a type error.

Author: That's not really our issue, IDEs complain when you leave out the comma as far as I know.
